nemo/collections/nlp/data/machine_translation/machine_translation_dataset.py

```diff
@@ -227,14 +227,21 @@ def pack_data_into_batches(self, src_ids, tgt_ids):
                 if batch_size * (src_len + tgt_len) > self.tokens_in_batch:
 
                     num_examples_to_split = len(batches[num_batches])
-                    batches_to_evict = 8 * ((num_examples_to_split - 1) // 8)
-
-                    if batches_to_evict == 0:
-                        batches_to_evict = num_examples_to_split
-
-                    batches.append(batches[num_batches][batches_to_evict:])
-                    batches[num_batches] = batches[num_batches][:batches_to_evict]
-                    batch_size = num_examples_to_split - batches_to_evict
+                    batches_to_keep = 8 * ((num_examples_to_split - 1) // 8)
```

Comment:
Can we rename `batches_to_keep` even a little further? I'd suggest `num_examples_to_keep`, since they're not actually batches.